yetanotherco/aligned_layer
# feat: rust sdk

# Description

- This PR creates an SDK for Aligned with 3 methods: **submit**, **verify_proof_onchain** and **get_verification_key_commitment**.

# Test the SDK
- Start anvil with block time, the aggregator, the operator and the batcher.
- Inside the `aligned-sdk` directory, run `cargo test`.
- Whenever the test freezes, run `make batcher_send_infinite_groth16` and keep sending proofs in the background.

# Test the client that now uses the SDK
- Run `make install_aligned_compiling`.
- Run `which aligned` and make sure you are using the `.cargo/bin/aligned` version.
- Make sure the client works as expected, run `Submit`, `VerifyProofOnChain` and `GetVerificationKeyCommitment`.

# Important

- I made an minimum example using the SDK, I will add it in another PR because I need the SDK merged in main.
- The client won't work with the batcher deployed on Testnet yet. We need to re-deploy the batcher for it to send the current version to the client.
